frontend/src/api/api.ts

```diff
@@ -0,0 +1,43 @@
+export const api = {
+  baseURL: '/api/rest/',
+  headers: {
+    Accept: 'application/json',
+  },
+  async get(url: string, params: Record<string, any> = {}) {
+    const query = new URLSearchParams(params).toString();
+    const response = await fetch(`${this.baseURL}${url}?${query}`, {
+      headers: this.headers,
+    });
+
+    if (!response.ok) {
+      throw new Error(`Error fetching data from ${url}`);
+    }
+
+    return response.json();
```

Comment:
Fetch method returns a promise, so you have to wait for the JSON - return await response.json();

Reply:
yeah but.
The response.json() method itself returns a Promise that resolves with the result of parsing the body text as JSON. Therefore, it's not necessary to use await again when returning response.json(). The await keyword is already used before fetch to wait for the Promise that fetch returns. When response.json() is returned, it will return a Promise that will be awaited where the get or post method is called.